modules/develop/pages/data-transforms/build.adoc

```diff
@@ -0,0 +1,655 @@
+  if ok {
+    fmt.Printf("Output topic 1: %s\n", outputTopic1)
+  } else {
+    fmt.Println("Only one output topic is set")
+  }
+
+	transform.OnRecordWritten(doNothing)
+}
+
+// Required callback, even if it does nothing
+func doNothing(event transform.WriteEvent, writer transform.RecordWriter) error {
+	return nil
+}
+----
+--
+Rust::
++
+--
+[source,rust]
+----
+use redpanda_transform_sdk::*;
+use std::env;
+use log::error;
+
+fn main() {
+  // Set up logging
+  env_logger::init();
+
+  // Access built-in environment variables
+  match env::var("REDPANDA_INPUT_TOPIC") {
+    Ok(input_topic) => println!("Input topic: {}", input_topic),
+    Err(_) => println!("An input topic is not set"),
+  }
+
+  match env::var("REDPANDA_OUTPUT_TOPIC_0") {
+    Ok(output_topic_0) => println!("Output topic 0: {}", output_topic_0),
+    Err(_) => println!("An output topic is not set"),
+  }
+
+  match env::var("REDPANDA_OUTPUT_TOPIC_1") {
+    Ok(output_topic_1) => println!("Output topic 1: {}", output_topic_1),
+    Err(_) => println!("Only one output topic is set"),
+  }
+
+  on_record_written(do_nothing);
+}
+
+// Required callback, even if it does nothing
+fn do_nothing(_event: WriteEvent, _writer: &mut RecordWriter) -> anyhow::Result<()> {
+  Ok(())
+}
+----
+--
+======
+
+=== Connect to the Schema Registry
+
+You can use the Schema Registry client library to read and write schemas as well as serialize and deserialize records. This client library is useful when working with schema-based topics in your data transforms.
+
+See also:
+
+- xref:manage:schema-reg/schema-reg-overview.adoc[].
+- xref:reference:data-transforms/golang-sdk.adoc[Go Schema Registry client reference]
+- xref:reference:data-transforms/rust-sdk.adoc[Rust Schema Registry client reference]
+
+==== Initialize the Schema Registry client
+
+First, initialize the Schema Registry client. This client will interact with the local Redpanda Schema Registry.
+
+[tabs]
+======
+Go::
++
+--
+[source,go]
+----
+package main
+
+import (
+  "context"
+  "fmt"
+	"github.com/redpanda-data/redpanda/src/transform-sdk/go/transform"
+	"github.com/redpanda-data/redpanda/src/transform-sdk/go/transform/sr"
+)
+
+func main() {
+  client := sr.NewClient()
+	transform.OnRecordWritten(doNothing)
+
+  // Use the client...
+}
+
+// Required callback, even if it does nothing
+func doNothing(event transform.WriteEvent, writer transform.RecordWriter) error {
+	return nil
+}
+----
+--
+Rust::
++
+--
+[source,rust]
+----
+use redpanda_transform_sdk::*;
+use redpanda_transform_sdk_sr::{SchemaRegistryClient};
+
+fn main() {
+    // Initialize the Schema Registry client
+    let client = SchemaRegistryClient::new();
+    on_record_written(do_nothing);
+
+    // Use the client
+}
+
+// Required callback, even if it does nothing
+fn do_nothing(_event: WriteEvent, _writer: &mut RecordWriter) -> anyhow::Result<()> {
+  Ok(())
+}
+----
+--
+======
+
+==== Read schemas from the Schema Registry
```

Comment:
This is all awesome, but we should show people how to interact with schemas when transforms are being executed as that's generally what people would want. Here's the pattern I expect in psuedo code:

```
client: sr_client

my_schema = "..."
my_schema_id = -1

main()
  client = create_sr_client()
  my_schema_id = write_schema(my_schema)
  on_record_written(my_transform)

my_transform(record, writer)
  schema_id, value = get_schema_id(record.value())
  record_schema = sr_client.get_schema(schema_id)
  struct = deserialize(value, my_schema, record_schema) # Assuming compatibility
  new_struct = do_transform(struct)
  new_value = serialize(new_struct, my_schema)
  new_value = add_schema_id(new_value, my_schema_id)
  writer.write(Record(record.key(), new_value))
```

we can certainly make this experience better over time.

Reply:
worth noting that rust is not particularly great at "global instance of complicated (possibly) mutable thing". 🤷

Reply:
thanks! @rockwotj or @oleiman do you have a simple working example of this in Go and Rust?

For Go, something like

```go
package main

import (
  "context"
  "fmt"
  "github.com/redpanda-data/redpanda/src/transform-sdk/go/transform"
  "github.com/redpanda-data/redpanda/src/transform-sdk/go/transform/sr"
)

func main() {
  // Initialize the Schema Registry client
  client := sr.NewClient()

  // Register your transform function.
  transform.OnRecordWritten(doTransform)
}

func doTransform(event transform.WriteEvent, writer transform.RecordWriter) error {
  // Lookup schema by ID
  schema, err := client.LookupSchemaById(1)
  if err != nil {
    panic(err)
  }

  fmt.Println("Schema:", schema.Schema)
  return writer.Write(event.Record())
}
```

I'm not sure how to get a reference to the global client.

Reply:
>working examples...?

Qualified yes:
- [golang](https://github.com/redpanda-data/redpanda/blob/dev/src/transform-sdk/go/transform/internal/testdata/schema-registry/transform.go) 
- [rust](https://github.com/redpanda-data/redpanda/blob/dev/src/transform-sdk/rust/examples/schema_registry/main.rs)

Qualified in the sense that these were written for integration testing and might not be suitable as docs-facing examples w/o some modification. Each exhibits a possible pattern for initializing and consuming an SR client though.

Reply:
Let's leave the examples for now and before we release 24.2 we can update them to show how to use JSON schemas.

Reply:
Since that's a new feature, we might want to showcase it here